### ISSUE TITLE
Create comparison guide with other sparsity methods

### ISSUE THREAD

**Issue by MaxGhenis** [PolicyEngine/L0#16]
## Description
Create a comprehensive guide comparing L0 with other sparsity/pruning methods.

### Method Comparisons
| Method | Type | Advantages | Disadvantages | When to Use |
|--------|------|------------|---------------|-------------|
| L0 | Learnable gates | Differentiable, adaptive | Complex, slower training | When you need learned sparsity |
| L1 | Weight penalty | Simple, well-understood | Rarely exact zeros | Feature selection |
| Magnitude Pruning | Post-training | Fast, simple | Not adaptive | Quick compression |
| Lottery Tickets | Iterative | High accuracy | Very expensive | Research/best accuracy |
| Dropout | Random | Prevents overfitting | Not true sparsity | Regularization only |

### Code Examples
```python
# Compare different methods on same task
results = {}

# L0
model_l0 = train_with_l0(base_model.clone())
results['L0'] = evaluate(model_l0)

# L1
model_l1 = train_with_l1(base_model.clone())  
results['L1'] = evaluate(model_l1)

# Magnitude pruning
model_mag = magnitude_prune(trained_model.clone())
results['Magnitude'] = evaluate(model_mag)

plot_comparison(results)
```

### Decision Tree
- Need differentiable sparsity? → L0
- Need feature selection? → L0/L1  
- Need quick compression? → Magnitude pruning
- Need best accuracy? → Lottery tickets/iterative
